### DISCUSSION TITLE
Pulsar OAuth2 base 64

### DISCUSSION THREAD

**Discussion by adavilag0600** [majusko/pulsar-java-spring-boot-starter#308] [Q&A] [answered]
I notieced there is option to use base64 to pass the JSON credentials file, do you have any example of it?

pulsar.oauth2-credentials-url - URL to a JSON credentials file. Support the following pattern formats: file:///path/to/file, file:/path/to/file or data:application/json;base64,<base64-encoded value>

I tried but in Pulsar 2.10 the Java APi only allows URL as parameter.

**Comment by majusko** [comment, marked as answer]
Sure, it's just extending the basic setup that the base Java library developed by Pulsar is using. There are nice docs about it. https://pulsar.apache.org/docs/2.10.x/security-oauth2/
